Platon/backend/app/follow_system/views.py

```diff
@@ -492,5 +505,118 @@ def delete(user_id, self):
         else:
             return make_response(jsonify({'error': 'Input Format Error'}), 400)
 
+@follow_system_ns.route("/report")
+class ReportAPI(Resource):
+    '''
+        Endpoints for Report Model.
+    '''
+
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: ' Database Connection Error',
+                        404: 'Not found'})
+    @api.response(200, 'Success', report_list_model)
+    @api.expect(get_report_parser)
+    @login_required
+    @follow_required(param_loc='args', requested_user_id_key='reported_user_id')
+    def get(user_id, self):
+        '''
+            Get Reports of User.
+        '''
+        form = GetReportsForm(request.args)
+        if form.validate():
+            try:
+                user_reports = Reports.query.filter(
+                    Reports.reported_user_id == form.reported_user_id.data).all()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+            if len(user_reports) == 0:
+                return make_response(jsonify({'result': []}), 200)
+            return_list = []
+            for report in user_reports:
+                try:
+                    reporter = User.query.filter(User.id == report.owner_id).first()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+                if reporter is None:
+                    return make_response(jsonify({'error': 'Reporter is None, smth wrong'}), 500)
+                return_list.append({
+                    "report_id": report.id,
+                    "owner_id": report.owner_id,
+                    "reported_user_id": report.reported_user_id,
+                    "text": report.text
+                })
+            # Pagination functionality
+            number_of_pages = 1
+            if form.page.data is not None and form.per_page.data is not None:
+                per_page = form.per_page.data
+                number_of_pages = math.ceil(len(return_list) / per_page)
+                # Assign the page index to the maximum if it exceeds the max index
+                page = form.page.data if form.page.data < number_of_pages else number_of_pages - 1
+                return_list = return_list[page * per_page:(page + 1) * per_page]
+            return make_response(jsonify({'number_of_pages': number_of_pages, 'result': return_list}), 200)
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
```

Comment:
Because we do not have an admin user type. Who will use this endpoint to get the reports of a user?

Reply:
I thought, in the frontend part they might want to display it since we don't have admin there is nothing left to do with the reported users.

Reply:
I do not have any idea about where they can show these reports. Maybe they can be stored to delete a report that I want to delete.

Reply:
In addition I think it can be beter to give more information(Like name, surname initials) about users according to the place that we will use.

Reply:
You are right, i will change the response model of the Report class. And yes, it makes sense to show them in the reporter and reported user profiles so they can view/review them.